src/deployment/deploy.py

```diff
@@ -371,12 +396,31 @@ def deploy_template(self) -> None:
         expiry = (datetime.now(TZ_UTC) + timedelta(days=365)).strftime(
             "%Y-%m-%dT%H:%M:%SZ"
         )
+
+        if self.multi_tenant_domain:
+            # clear the value in the Issuer Url field:
+            # https://docs.microsoft.com/en-us/sharepoint/dev/spfx/use-aadhttpclient-enterpriseapi-multitenant
+            app_func_audience = "https://%s/%s" % (
+                self.multi_tenant_domain,
+                self.application_name,
+            )
+            app_func_issuer = ""
+            multi_tenant_domain = {"value": self.multi_tenant_domain}
+        else:
+            app_func_audience = "https://%s.azurewebsites.net" % self.application_name
+            tenant_oid = str(self.cli_config["authority"]).split("/")[-1]
+            app_func_issuer = "https://sts.windows.net/%s/" % tenant_oid
+            multi_tenant_domain = {"value": ""}
+
         params = {
+            "app_func_audience": {"value": app_func_audience},
             "name": {"value": self.application_name},
             "owner": {"value": self.owner},
             "clientId": {"value": self.results["client_id"]},
             "clientSecret": {"value": self.results["client_secret"]},
+            "app_func_issuer": {"value": app_func_issuer},
             "signedExpiry": {"value": expiry},
+            "multi_tenant_domain": multi_tenant_domain,
```

Comment:
seems like this key should not be added when the multi_tenant_domain is not defined. 
With the current code the value will be an empty string.
i propose you don't set the value when there are no multitenant domain and conditionally add the key like this 
``` Python
{
            "multi_tenant_domain" if multi_tenant_domain: multi_tenant_domain
}
```

same goes for `multi_tenant_domain`